metaswitch/common/generate_stats_csv.py

```diff
@@ -1,293 +1,296 @@
+    )
+
+MibData = collections.namedtuple("MibData", ["source_file",
+                                             "mib_table_description",
+                                             "mib_field_description",
+                                             "oid"])
+CsvData = collections.namedtuple("CsvData", ["version_introduced",
+                                             "calculation_type",
+                                             "reset_trigger",
+                                             "reset_trigger_detail",
+                                             "data_type",
+                                             "index_fields",
+                                             "index_field_values",
+                                             "field_size",
+                                             "field_units",
+                                             "aggregation_mechanism"])
+KeyData = collections.namedtuple("KeyData", ["mib_table_name",
+                                             "mib_field_name"])
+
+
+def main():
+    """Main entry point for the script."""
+    setup_logging()
+    args = parse_args()
+
+    parsed_mibs = parse_mib_files(args['mib_files'])
+    try:
+        parsed_csv = parse_csv_file(args['csv_file'])
+    except IOError:
+        sys.exit("Failed to open the CSV file %s." % args['csv_file'])
+    merged_entries = merge_csv_with_mibs(parsed_csv, parsed_mibs)
+
+    write_csv(merged_entries, args['output_file'])
+
+
+def setup_logging():
+    """Set up basic logging."""
+    # TODO write log level lower than warnings to file if log level set
+    # lower than warning.
+    level = os.getenv('LOG_LEVEL', 'WARNING')
+    logging.basicConfig(level=getattr(logging, level))
+
+
+def parse_args():
+    """Get the command-line arguments."""
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument('mib_files', metavar='MIB', nargs='+',
+                        help='The absolute path(s) of the input MIB(s).')
+    parser.add_argument('csv_file', metavar='CSV',
+                        help='The absolute path of the input CSV.')
+    parser.add_argument('--output-file', default='./output.csv',
+                        help='Optional output file name (defaults to '
+                        'output.csv in the current directory).')
+    args = vars(parser.parse_args())
+    logger.debug("Command-line arguments: %s", args)
+    return args
+
+
+def parse_mib_files(mib_files):
+    """Parse the specified MIB files.
+
+    `mib_files` should be an iterator of absolute paths to ASN.1 MIB files on
+    disk.
+    Returns a dict keyed by (MIB table name, MIB field name) with values
+    (Source File, MIB table description, OID, MIB field description).
+    If there are any duplicate keys in the provided MIB files, the initial
+    value gets overwritten by any occurrence in a MIB file that was provided
+    after the MIB file with the first occurrence on the command line.
```

Comment:
Is that the right behaviour? If it is, it's worth documenting that that's what the script does in it's docstring, and adding a comment here explaining why it's the right thing to do.